src/main/rust/bridge/loadable.rs

```diff
@@ -12,13 +14,16 @@ use url::Url;
 use crate::bridge::ToProtobuf;
 use crate::protobuf::stremio::core::models;
 
-impl ToProtobuf<models::loadable_catalog::Content, ResourceRequest>
+impl ToProtobuf<models::loadable_catalog::Content, (&Ctx, ResourceRequest)>
     for Loadable<Vec<MetaItemPreview>, ResourceError>
 {
-    fn to_protobuf(&self, request: &ResourceRequest) -> models::loadable_catalog::Content {
+    fn to_protobuf(
+        &self,
+        (ctx, request): &(&Ctx, ResourceRequest),
```

Comment:
signatures like this should look like:
&(&Ctx, &ResourceRequest)

the impl signature should be like:
impl ToProtobuf<models::loadable_catalog::Content, (&Ctx, &ResourceRequest)>

refactor all impl ToProtobuf to minimize mem copy and use references where its possible and needed